mantidproject/mantid
# Fix ICat Login for new Reflectometry GUI on Workbench

**Description of work.**
Previously searching without logging in for ICat on the workbench would cause a segfault. This is no longer the case, I changed the basic behaviour of the login format, to accommodate logging in pre-search with the added bonus of it remembering your username.

**To test:**
- Open Workbench
- Open new GUI (ISIS Reflectometry)
- Select INTER and enter search number: 1120015
- Hit search and login
- The search results should now appear



Fixes #25970. 



*This does not require release notes* because it is an in-progress feature that has release notes.


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
